### PR TITLE
Prevent Form Submission

### PR DESCRIPTION
**Pull request by morecallan**

### CODE REVIEW

File: Tweeter/Scripts/app.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    $scope.checkToSeeIfUsernameExists = function () {
+        var url = `/api/TwitUsername?usernameCandidate=${$scope.usernameCandidate}`
+
+        return $q(function (resolve, reject) {
+            $http.get(url).then(function (response) {
+                var data = response.data;
+                if (data.exists) {
+                    $scope.userNameExistsCode = 1;
+                    $("#register").submit(function (e) {
```
Comment (katerebekah) [inline Tweeter/Scripts/app.js:15]: Why are you submitting than immediately un-submitting?

File: Tweeter/Scripts/app.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+﻿var app = angular.module('Tweeter', []);
+
+app.controller('Register', function ($scope, $q, $http) {
+
+    $scope.userNameExistsCode = null;
```
Comment (katerebekah) [inline Tweeter/Scripts/app.js:5]: Is there a reason you used an int for this rather than a bool?

**Review by katerebekah — COMMENTED** [review]
Nice angular implementation 👍

File: Tweeter/Scripts/app.js (hunk 2 above)
Comment (morecallan) [reply Tweeter/Scripts/app.js:5]: @katerebekah
This is simply to use a falsy value of `null` if the function hasn't run yet just for the ng-class.
ng-class:{'disabled': (!userNameExistsCode || userNameExistsCode == 1)}

**Pull request closed without merging**